lib/ClangImporter/ImportDecl.cpp

```diff
@@ -1593,6 +1597,7 @@ buildSubscriptGetterDecl(ClangImporter::Implementation &Impl,
   if (auto objcAttr = getter->getAttrs().getAttribute<ObjCAttr>())
     thunk->getAttrs().add(objcAttr->clone(C));
   thunk->setIsObjC(getter->isObjC());
+  thunk->setIsDynamic(getter->isDynamic());
```

Comment:
This doesn't make sense to me. The getter is always an imported declaration, and the thunk is always a thunk and isn't exposed to Objective-C itself, even when the subscript is on a class.

Reply:
Since it was already copying the @objc bit from the getter I'm going to leave the cargo cult in and copy the dynamic bit over too. For imported declarations the two are essentially isomorphic, and if I set it to 'false' I get test failures.

Reply:
To close the loop here, I was confused about what this code was actually doing. Copying `@objc` and `dynamic` is the correct thing to do. (And maybe I'll rename the local variable to something other than "thunk" later.)